packages/next/src/server/lib/start-server.ts

```diff
@@ -279,7 +281,8 @@ export async function startServer(
           // This is the render worker, we keep the process alive
           console.error(err)
         }
-        process.on('exit', (code) => cleanup(code))
```

Comment:
https://nodejs.org/api/process.html#event-exit

> Listener functions **must** only perform **synchronous** operations. The Node.js process will exit immediately after calling the `'exit'` event listeners causing any additional work still queued in the event loop to be abandoned.

Once `process.exit` is called we don't really have a chance to clean up. Best we could do is log a message, but not sure that's very useful.

Also calling the same cleanup code, now that there's a log, was resulting in the log printing twice after a signal was received.